src/esp/gfx/PrimitiveIDTexturedShader.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <Corrade/Containers/EnumSet.h>
+
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+
+#include "esp/core/esp.h"
+
+namespace esp {
+namespace gfx {
+
+class PrimitiveIDTexturedShader : public Magnum::GL::AbstractShaderProgram {
+ public:
+  /**
+   * @brief Constructor
+   */
+  explicit PrimitiveIDTexturedShader();
+
+  //! @brief vertex positions
+  typedef Magnum::GL::Attribute<0, Magnum::Vector4> Position;
+  //! @brief vertex colors
+  typedef Magnum::GL::Attribute<3, Magnum::Vector3> Color;
+
+  //! Color attachment location per output type
+  enum : uint8_t {
+    //! color output
+    ColorOutput = 0,
+    //! object id output
+    ObjectIdOutput = 1
+  };
+
+  /**
+   * @brief Set transformation and projection matrix
+   * @return Reference to self (for method chaining)
+   */
+  PrimitiveIDTexturedShader& setTransformationProjectionMatrix(
+      const Magnum::Matrix4& matrix) {
+    setUniform(uniformLocation("transformationProjectionMatrix"), matrix);
```

Comment:
@mosra:
As you suggested in your comments at #132, do you think we should cache the uniform location of the "transformationProjectionMatrix" in the constructor?
The same for the "texSize".

Reply:
Here I did just the most brainless copy of the original shader, trying to introduce as little changes (and potential bugs) as possible :) I'll be opening a cleanup PR for this shader (fixing also the webgl TODO) right once this is merged.